### PR TITLE
Use alternate image as fallback for creation date

### PR DESCRIPTION
**Pull request by rc-gr**
This happens when the saved image is valid, but not present among the generated images in the prompt URL.

This behavior is also present when accessing the saved image manually in the collection via the browser. ie. The thumbnail shows the correct image, but is not among the images in the prompt URL.

From initial testing, the creation date and prompt of the alternate image, which is the first of the generated images in the prompt URL, seems to line up with the saved one.

In other words, this handles cases Image X is accessed with the correct and valid thumbnail and image URL, but prompt URL contains images A,(B),(C),(D) with matching prompt and date.

### CODE REVIEW

**Comment by Richard-Weiss** [comment]
@rc-gr I'll take a look into that on the weekend, but I'm still not completely understanding the issue.  
Can you elaborate more or show some specific examples?

**Comment by rc-gr** [comment]
Referring to the observation that you mentioned [here](https://github.com/Richard-Weiss/Bing-Creator-Image-Downloader/commit/178fdd2962a1973e2d7591927c239f4193340282?short_path=b335630#):

> Because of how the images are accessed there may be some images that are duplicated or **_images that appear that weren't in the text file_**.

Suppose you save Image X which is initially present on Prompt Page X. In the collections where it's saved, it reflects as such with corresponding Thumbnail X.

Some time later, when you try to access Image X from your collections, Image Y is shown instead on Prompt Page X, along with 1-3 other incorrect images from the prompt. But the prompt and generation date shown on the page is still correct.

What you probably observed back then was Image Y being downloaded instead of Image X, because Image X no longer exists on Prompt Page X.

Currently, without the commit, the program would error out when the above occurs, because Image X's `decoded_image_id` would not be found among the `imageId`s of `images`. This causes `resp_image` on line 271 to be an empty array.

With this commit, the program will sidestep the issue by taking the creation date from Image Y, which is valid as the first image of the set of images on Prompt Page X, which is similar to Image X's date.

Note that this will not affect the downloaded image or thumbnail, which are valid and correct regardless.

**Comment by Richard-Weiss** [comment]
@rc-gr Is this something that can actually happen? I mean it sounds like a nice workaround, but I don't see how the imageId or imageSetId could change for a given image in the collection. The mediaUrl might, but not the pageUrl or any of these fixed values.  
Do you maybe have an image that exhibits this behavior so I can reproduce?

**Comment by Richard-Weiss** [comment]
@rc-gr I get what you mean now. It doesn't really matter which one of these images we take as the creation date is the same anyways.  
I just find it weird that the request works but the image won't be returned in the response.  
Anyways, looks good to me. 🙂
